net.solarnetwork.node.datum.energymeter.mock/src/net/solarnetwork/node/datum/energymeter/mock/MockEnergyMeterDatumSource.java

```diff
@@ -45,6 +49,12 @@
 	private Integer realPow;
 	private Integer reacPow;
 	private Float powfac;
+	private Long watthours;
+	private Long wattmillis;
+
+	private Long lastsample = null;
```

Comment:
In other code there are `lastSample` fields that are actual sample data objects, so I was mislead by this property name when I saw it was just a time value. It would be good to sick to that convention here, so instead of storing the sample time and energy fields, just keep the entire `GeneralNodeACEnergyDatum` instance as `lastsample`. That will then have the energy and anything else you need to use for deriving properties like power over time.